src/tools/schema2graph.ts

```diff
@@ -56,6 +57,8 @@ export class SchemaNode {
 
   readonly sources: SchemaSource[] = [];
 
+  hash: string | undefined;
```

Comment:
Why `undefined` over `null`? This is not a suggestion, I'm curious why we would prefer one over the other, in general.

Reply:
I could do:
```
hash: string | null = null;
```

It just that undefined is the value you get if you don't assign anything. I don't actually have a preference of one vs the other. Null is something you have to assign explicitly, undefined is what you get if no value is provided.

I will change to null, so that it's more explicit.